dv/uvm/icache/doc/ibex_icache_dv_plan.md

```diff
@@ -1,99 +1,142 @@
 ---
-title: "IBEX_ICACHE DV Plan"
+title: "Ibex ICache DV Plan"
 ---
 
 ## Goals
 * **DV**
-  * Verify all IBEX_ICACHE IP features by running dynamic simulations with a SV/UVM based testbench
-  * Develop and run all tests based on the [testplan](#testplan) below towards closing code and functional coverage on the IP and all of its sub-modules
+  * Verify all Ibex ICache IP features by running dynamic simulations with a SV/UVM based testbench
+  * Develop and run all tests based on the [testplan](#testplan) below towards closing code and functional coverage.
 * **FPV**
-
-## Current status
-<!-- TODO for now -->
-* [Design & verification stage]({{< relref "doc/project/hw_dashboard" >}})
-  * [HW development stages]({{< relref "doc/project/hw_stages" >}})
-* [Simulation results](https://reports.opentitan.org/hw/ip/ibex_icache/dv/latest/results.html)
+  * [TBD]
 
 ## Design features
-For detailed information on IBEX_ICACHE design features, please see the [IBEX_ICACHE technical specification]({{< relref "doc/icache.rst" >}}).
+
+The ICache design is documented in the [Instruction Cache](https://ibex-core.readthedocs.io/en/latest/icache.html) section of the Ibex User Manual.
 
 ## Testbench architecture
-IBEX_ICACHE testbench has been constructed based on the [DV_LIB testbench architecture]({{< relref "vendor/lowrisc_ip/dv_lib/" >}}).
```

Comment:
I think a short blurb referencing the dv_lib tb architecture should be kept in the dv plan, @sriyerg what do you think?

Reply:
Who is the target audience for this document? I decided it was people who want to know whether the icache works or not. I shouldn't imagine those people care about our class structure. People who need to actually understand the test code will presumably look at the source, at which point it will be obvious what it's derived from.

Reply:
Ok. I'm not really convinced, but I don't mind putting the auto-generated text back if you believe it's important. I'll push up a patch with this in a few minutes.

Reply:
> @sriyerg I understand your note about the testplan, but don't think it really makes sense to write that without a rough block diagram for the testbench, and I'm blocked on implementing the testbench until you and Udi say you're ok with the basic design - I don't want to write thousands of lines of code and then have to rework them all.
> 
> If you think this DV plan is a reasonable start (and describes a sane framework), maybe the best thing to do is to merge this and hash out details later, in parallel with actually building the thing.

A testplan captures what all features are tested. This only requires the design spec to be available as a starting point. You need all of that information first, to know how the testbench needs to be put together. The "what" dictates the "how". The suggestion I was trying to make was to focus on the testplan more. As an example, from the AES testplan review, it was found that it was missing NIST vector testing (which required additional testbench components to be added, and hence DV plan updated). Also, it is reasonable to expect that once you start developing the code, you may realize that how you had originally envisioned the scoreboard will work for example, needs to complete change for whatever reason.

A rough block diagram for the testbench in the DV plan at this stage is highly recommended. However the nitty gritty details on how the scoreboard will work for example, is not required at this stage (but definitely welcome!). The actual implemented code will go through PR reviews anyway. 

I am fine with this landing as-is. We can modify the content as we progress through the development.

Reply:
Ok, thanks for the explanation.